File: w09/d03/morgan_neiman/mail_objects_app/app/assets/javascripts/mailbox.js
function Mailbox() {
  this.mailItems = [];
}

Mailbox.prototype.acceptMail = function(mail) {
  if(mail.constructor == Postcard || mail.constructor == Package ) {
    this.mailItems.push(mail);
  }
  else {
    throw "must be mail item";
  }
};

Mailbox.prototype.openMail = function() {

  this.mailItems.shift();

};

Mailbox.prototype.toString = function() {
  var postcards = 0;
  var packages = 0;
  for( i = 0; i < this.mailItems.length; i++ ) {
    if( this.mailItems[i].constructor == Postcard ) {
      postcards++;
    }
    else {
      packages++;
    }
  }
  return "The mailbox has " + postcards + " postcards and " + packages + " packages";
};

File: w09/d01/ana_giraldo-wingler/pair_testing/mailbox.rb
require_relative 'mail_item'

class Mailbox < MailItem
  attr_accessor :mail_items

  def initialize
    @mail_items = []
  end

  def accept_mail(mail)
    # 'mail' must be a mail object in order to be pushed into mail_items
    @mail_items << mail if mail.class < MailItem
  end

  def open_mail
    @mail_items.shift
  end

  def to_s
    num_postcards = 0
    num_packages = 0

    @mail_items.each do |item|
      if item.class == PostCard
        num_postcards += 1
      else
        num_packages += 1
      end
    end

  return "The mailbox has #{num_postcards} postcards and #{num_packages} packages."
  end

end